src/terminal.rs

```diff
@@ -236,7 +236,15 @@ impl<H: Hasher> Terminal<H> {
             .get(u_hash)
             .expect("Unit to be added to dag must be in store")
             .clone();
-        self.post_insert.iter().for_each(|f| f(u.clone()));
+        if self
+            .post_insert
+            .iter()
+            .map(|f| f(u.clone()))
+            .any(|result| result.is_err())
+        {
+            debug!(target: "AlephBFT-terminal", "{:?} error in post_insert hook, closing", self.node_id);
+            return Err(());
```

Comment:
This significantly changes the semantics of this method – are we sure skipping the operations below when a hook fails is really what we want to do? I would expect these hooks should be executed _after_ the calls below.

Reply:
All the hooks are crucial for the execution so if one fails we return error and close the party.